packages/client/tests/functional/issues/8832/tests.ts

```diff
@@ -0,0 +1,106 @@
+      const ids = Array.from({ length }, (_, i) => i + 1)
+      const data = ids.map((id) => ({ id }))
+      await prisma.tag.createMany({
+        data,
+      })
+      return ids
+    }
+
+    async function getTagsFindManyIn(ids: number[]): Promise<unknown[]> {
+      const tags = await prisma.tag.findMany({
+        where: {
+          id: {
+            in: ids,
+          },
+        },
+      })
+      return tags
+    }
+
+    async function getTagsFindManyInclude(): Promise<unknown[]> {
+      const tags = await prisma.tag.findMany({
+        include: {
+          posts: true,
+        },
+      })
+      return tags
+    }
+
+    async function clean() {
+      const cleanPrismaPromises = [prisma.tagsOnPosts.deleteMany(), prisma.post.deleteMany(), prisma.tag.deleteMany()]
+      await prisma.$transaction(cleanPrismaPromises)
+    }
+
+    describe('issue #8832', () => {
+      beforeEach(async () => {
+        await clean()
+      }, 10_000)
+
+      test('should succeed when "in" has 32766 or less ids', async () => {
+        const n = 32766
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32766 or less records', async () => {
+        const n = 32766
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should fail with `value too large to transmit` when "in" has 32767+ ids', async () => {
+        expect.assertions(3)
+        const n = 32767
+        const ids = await createTags(n)
+
+        try {
+          await getTagsFindManyIn(ids)
+        } catch (e) {
+          const error = e as Error & { code: number; meta?: unknown }
+          expect(error.message).toContain('Assertion violation on the database: `value too large to transmit`')
+          expect(error.code).toBe('P2034')
+          expect(error.meta).toMatchObject({
+            database_error: 'value too large to transmit',
+          })
+        }
+      })
+
+      test('should fail with `value too large to transmit` when "include" involves 32767+ records', async () => {
+        expect.assertions(3)
+        const n = 32767
+        const ids = await createTags(n)
+
+        try {
+          await getTagsFindManyInclude()
+        } catch (e) {
+          const error = e as Error & { code: number; meta?: unknown }
+          expect(error.message).toContain('Assertion violation on the database: `value too large to transmit`')
+          expect(error.code).toBe('P2034')
+          expect(error.meta).toMatchObject({
+            database_error: 'value too large to transmit',
+          })
+        }
+      })
+    })
+  },
+  {
+    optOut: {
+      from: ['mysql', 'sqlserver', 'cockroachdb', 'sqlite', 'mongodb'],
+      reason: 'we have only captured this issue with postgres',
+    },
```

Comment:
```suggestion
    },
    skipDataProxy: {
      runtimes: ['node', 'edge'],
      reason: "Mini-Proxy can't currently handle big queries, un-skip when it starts using QE server instead of CLI",
    },
```

Reply:
Do we have an issue for this @aqrln? We should...

Reply:
(Completely skipping Data Proxy on this means that a huge chunk of real world usage is not tested for Data Proxy Client - which makes me slightly nervous. (Including the new error that Alberto is introducing and trying to test here)